File: php/lib/Stakkd/model/DemographicsAppendResponseDemographicsPersonFinances.php
<?php
/**
 * DemographicsAppendResponseDemographicsPersonFinances
 *
 * PHP version 7.3
 *
 * @category Class
 * @package  OpenAPI\Client
 * @author   OpenAPI Generator team
 * @link     https://openapi-generator.tech
 */

/**
 * Stakkd API
 *
 * # Introduction The Stakkd platform provides comprehensive data services that meet the business objectives of organizations ranging from Fortune 500 companies to startups. Our clients work in an array of industries, including insurance, financial, legal, travel, hospitality, retail, health, media, and telecommunications.  The Stakkd API is documented using the Open API 3.0.1 standard.  All endpoints are available at https://api.stakkd.io/v1. Detailed documentation for each endpoint is below, including the full URL for the endpoint.  # Account Types The API supports two different account types, each with different permissions.  ## Paid Account Paid accounts provide unlimited access to all of the URIs.  ## Trial Account Trial Accounts have a limited number of queries to the APIs. Keep in mind that queries which do not return results still count against the Trial Account query limits. Trial Accounts are temporary, but can be upgraded or extended by contacting [support@stakkd.io](mailto:support@stakkd.io)
 *
 * The version of the OpenAPI document: v1
 * Generated by: https://openapi-generator.tech
 * OpenAPI Generator version: 5.4.0
 */

/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

namespace OpenAPI\Client\Stakkd.model;

use \ArrayAccess;
use \OpenAPI\Client\ObjectSerializer;

/**
 * DemographicsAppendResponseDemographicsPersonFinances Class Doc Comment
 *
 * @category Class
 * @package  OpenAPI\Client
 * @author   OpenAPI Generator team
 * @link     https://openapi-generator.tech
 * @implements \ArrayAccess<TKey, TValue>
 * @template TKey int|null
 * @template TValue mixed|null
 */
class DemographicsAppendResponseDemographicsPersonFinances implements ModelInterface, ArrayAccess, \JsonSerializable
{
    public const DISCRIMINATOR = null;

    /**
      * The original name of the model.
      *
      * @var string
      */
    protected static $openAPIModelName = 'demographicsAppendResponse_demographics_person_finances';

    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $openAPITypes = [
        'estimated_income' => 'string',
        'estimated_net_worth' => 'string',
        'is_charity_donar' => 'bool',
        'uses_credit_card' => 'bool'
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      * @phpstan-var array<string, string|null>
      * @psalm-var array<string, string|null>
      */
    protected static $openAPIFormats = [
        'estimated_income' => null,
        'estimated_net_worth' => null,
        'is_charity_donar' => null,
        'uses_credit_card' => null
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function openAPITypes()
    {
        return self::$openAPITypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function openAPIFormats()
    {
        return self::$openAPIFormats;
    }

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @var string[]
     */
    protected static $attributeMap = [
        'estimated_income' => 'estimatedIncome',
        'estimated_net_worth' => 'estimatedNetWorth',
        'is_charity_donar' => 'isCharityDonar',
        'uses_credit_card' => 'usesCreditCard'
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'estimated_income' => 'setEstimatedIncome',
        'estimated_net_worth' => 'setEstimatedNetWorth',
        'is_charity_donar' => 'setIsCharityDonar',
        'uses_credit_card' => 'setUsesCreditCard'
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'estimated_income' => 'getEstimatedIncome',
        'estimated_net_worth' => 'getEstimatedNetWorth',
        'is_charity_donar' => 'getIsCharityDonar',
        'uses_credit_card' => 'getUsesCreditCard'
    ];

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @return array
     */
    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * The original name of the model.
     *
     * @return string
     */
    public function getModelName()
    {
        return self::$openAPIModelName;
    }

    const ESTIMATED_INCOME_A = 'A';
    const ESTIMATED_INCOME_B = 'B';
    const ESTIMATED_INCOME_C = 'C';
    const ESTIMATED_INCOME_D = 'D';
    const ESTIMATED_INCOME_E = 'E';
    const ESTIMATED_INCOME_F = 'F';
    const ESTIMATED_INCOME_G = 'G';
    const ESTIMATED_INCOME_H = 'H';
    const ESTIMATED_INCOME_I = 'I';
    const ESTIMATED_NET_WORTH_A = 'A';
    const ESTIMATED_NET_WORTH_B = 'B';
    const ESTIMATED_NET_WORTH_C = 'C';
    const ESTIMATED_NET_WORTH_D = 'D';
    const ESTIMATED_NET_WORTH_E = 'E';
    const ESTIMATED_NET_WORTH_F = 'F';
    const ESTIMATED_NET_WORTH_G = 'G';
    const ESTIMATED_NET_WORTH_H = 'H';

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public function getEstimatedIncomeAllowableValues()
    {
        return [
            self::ESTIMATED_INCOME_A,
            self::ESTIMATED_INCOME_B,
            self::ESTIMATED_INCOME_C,
            self::ESTIMATED_INCOME_D,
            self::ESTIMATED_INCOME_E,
            self::ESTIMATED_INCOME_F,
            self::ESTIMATED_INCOME_G,
            self::ESTIMATED_INCOME_H,
            self::ESTIMATED_INCOME_I,
        ];
    }

    /**
     * Gets allowable values of the enum
     *
     * @return string[]
     */
    public function getEstimatedNetWorthAllowableValues()
    {
        return [
            self::ESTIMATED_NET_WORTH_A,
            self::ESTIMATED_NET_WORTH_B,
            self::ESTIMATED_NET_WORTH_C,
            self::ESTIMATED_NET_WORTH_D,
            self::ESTIMATED_NET_WORTH_E,
            self::ESTIMATED_NET_WORTH_F,
            self::ESTIMATED_NET_WORTH_G,
            self::ESTIMATED_NET_WORTH_H,
        ];
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     *                      initializing the model
     */
    public function __construct(array $data = null)
    {
        $this->container['estimated_income'] = $data['estimated_income'] ?? null;
        $this->container['estimated_net_worth'] = $data['estimated_net_worth'] ?? null;
        $this->container['is_charity_donar'] = $data['is_charity_donar'] ?? null;
        $this->container['uses_credit_card'] = $data['uses_credit_card'] ?? null;
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        $allowedValues = $this->getEstimatedIncomeAllowableValues();
        if (!is_null($this->container['estimated_income']) && !in_array($this->container['estimated_income'], $allowedValues, true)) {
            $invalidProperties[] = sprintf(
                "invalid value '%s' for 'estimated_income', must be one of '%s'",
                $this->container['estimated_income'],
                implode("', '", $allowedValues)
            );
        }

        $allowedValues = $this->getEstimatedNetWorthAllowableValues();
        if (!is_null($this->container['estimated_net_worth']) && !in_array($this->container['estimated_net_worth'], $allowedValues, true)) {
            $invalidProperties[] = sprintf(
                "invalid value '%s' for 'estimated_net_worth', must be one of '%s'",
                $this->container['estimated_net_worth'],
                implode("', '", $allowedValues)
            );
        }

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }


    /**
     * Gets estimated_income
     *
     * @return string|null
     */
    public function getEstimatedIncome()
    {
        return $this->container['estimated_income'];
    }

    /**
     * Sets estimated_income
     *
     * @param string|null $estimated_income The estimated income of the person, where: A = < $20, 000; B = $20, 000 - $29, 999; C = $30, 000 - $39, 999; D = $40, 000 - $49, 999; E = $50, 000 - $74, 999; F = $75, 000 - $99, 999; G = $100, 000 - $124, 999; H = $125, 000 - $149, 999; I = > $149,999.
     *
     * @return self
     */
    public function setEstimatedIncome($estimated_income)
    {
        $allowedValues = $this->getEstimatedIncomeAllowableValues();
        if (!is_null($estimated_income) && !in_array($estimated_income, $allowedValues, true)) {
            throw new \InvalidArgumentException(
                sprintf(
                    "Invalid value '%s' for 'estimated_income', must be one of '%s'",
                    $estimated_income,
                    implode("', '", $allowedValues)
                )
            );
        }
        $this->container['estimated_income'] = $estimated_income;

        return $this;
    }

    /**
     * Gets estimated_net_worth
     *
     * @return string|null
     */
    public function getEstimatedNetWorth()
    {
        return $this->container['estimated_net_worth'];
    }

    /**
     * Sets estimated_net_worth
     *
     * @param string|null $estimated_net_worth The estimated net worth of the person, where: A = < $5k; B = $5k - $19k; C = $20k - $49k; D = $50k - $80k; E = $81k - $99k; F = $100k - $249k; G = $250k - $499k; H = > $500k.
     *
     * @return self
     */
    public function setEstimatedNetWorth($estimated_net_worth)
    {
        $allowedValues = $this->getEstimatedNetWorthAllowableValues();
        if (!is_null($estimated_net_worth) && !in_array($estimated_net_worth, $allowedValues, true)) {
            throw new \InvalidArgumentException(
                sprintf(
                    "Invalid value '%s' for 'estimated_net_worth', must be one of '%s'",
                    $estimated_net_worth,
                    implode("', '", $allowedValues)
                )
            );
        }
        $this->container['estimated_net_worth'] = $estimated_net_worth;

        return $this;
    }

    /**
     * Gets is_charity_donar
     *
     * @return bool|null
     */
    public function getIsCharityDonar()
    {
        return $this->container['is_charity_donar'];
    }

    /**
     * Sets is_charity_donar
     *
     * @param bool|null $is_charity_donar Indicates whether the person is flagged as a charity donar.
     *
     * @return self
     */
    public function setIsCharityDonar($is_charity_donar)
    {
        $this->container['is_charity_donar'] = $is_charity_donar;

        return $this;
    }

    /**
     * Gets uses_credit_card
     *
     * @return bool|null
     */
    public function getUsesCreditCard()
    {
        return $this->container['uses_credit_card'];
    }

    /**
     * Sets uses_credit_card
     *
     * @param bool|null $uses_credit_card Indicates whether the person is flagged as a credit card user.
     *
     * @return self
     */
    public function setUsesCreditCard($uses_credit_card)
    {
        $this->container['uses_credit_card'] = $uses_credit_card;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param integer $offset Offset
     *
     * @return boolean
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param integer $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param integer $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }

    /**
     * Serializes the object to a value that can be serialized natively by json_encode().
     * @link https://www.php.net/manual/en/jsonserializable.jsonserialize.php
     *
     * @return mixed Returns data which can be serialized by json_encode(), which is a value
     * of any type other than a resource.
     */
    public function jsonSerialize()
    {
       return ObjectSerializer::sanitizeForSerialization($this);
    }

    /**
     * Gets the string presentation of the object
     *
     * @return string
     */
    public function __toString()
    {
        return json_encode(
            ObjectSerializer::sanitizeForSerialization($this),
            JSON_PRETTY_PRINT
        );
    }

    /**
     * Gets a header-safe presentation of the object
     *
     * @return string
     */
    public function toHeaderValue()
    {
        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}

File: csharp/src/Stakkd/Stakkd.model/DemographicsAppendResponseDemographicsPersonFinances.cs
/*
 * Stakkd API
 *
 * # Introduction The Stakkd platform provides comprehensive data services that meet the business objectives of organizations ranging from Fortune 500 companies to startups. Our clients work in an array of industries, including insurance, financial, legal, travel, hospitality, retail, health, media, and telecommunications.  The Stakkd API is documented using the Open API 3.0.1 standard.  All endpoints are available at https://api.stakkd.io/v1. Detailed documentation for each endpoint is below, including the full URL for the endpoint.  # Account Types The API supports two different account types, each with different permissions.  ## Paid Account Paid accounts provide unlimited access to all of the URIs.  ## Trial Account Trial Accounts have a limited number of queries to the APIs. Keep in mind that queries which do not return results still count against the Trial Account query limits. Trial Accounts are temporary, but can be upgraded or extended by contacting [support@stakkd.io](mailto:support@stakkd.io) 
 *
 * The version of the OpenAPI document: v1
 * Generated by: https://github.com/openapitools/openapi-generator.git
 */


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = Stakkd.Client.OpenAPIDateConverter;

namespace Stakkd.Stakkd.model
{
    /// <summary>
    /// DemographicsAppendResponseDemographicsPersonFinances
    /// </summary>
    [DataContract(Name = "demographicsAppendResponse_demographics_person_finances")]
    public partial class DemographicsAppendResponseDemographicsPersonFinances : IEquatable<DemographicsAppendResponseDemographicsPersonFinances>, IValidatableObject
    {
        /// <summary>
        /// The estimated income of the person, where: A &#x3D; &lt; $20, 000; B &#x3D; $20, 000 - $29, 999; C &#x3D; $30, 000 - $39, 999; D &#x3D; $40, 000 - $49, 999; E &#x3D; $50, 000 - $74, 999; F &#x3D; $75, 000 - $99, 999; G &#x3D; $100, 000 - $124, 999; H &#x3D; $125, 000 - $149, 999; I &#x3D; &gt; $149,999.
        /// </summary>
        /// <value>The estimated income of the person, where: A &#x3D; &lt; $20, 000; B &#x3D; $20, 000 - $29, 999; C &#x3D; $30, 000 - $39, 999; D &#x3D; $40, 000 - $49, 999; E &#x3D; $50, 000 - $74, 999; F &#x3D; $75, 000 - $99, 999; G &#x3D; $100, 000 - $124, 999; H &#x3D; $125, 000 - $149, 999; I &#x3D; &gt; $149,999.</value>
        [JsonConverter(typeof(StringEnumConverter))]
        public enum EstimatedIncomeEnum
        {
            /// <summary>
            /// Enum A for value: A
            /// </summary>
            [EnumMember(Value = "A")]
            A = 1,

            /// <summary>
            /// Enum B for value: B
            /// </summary>
            [EnumMember(Value = "B")]
            B = 2,

            /// <summary>
            /// Enum C for value: C
            /// </summary>
            [EnumMember(Value = "C")]
            C = 3,

            /// <summary>
            /// Enum D for value: D
            /// </summary>
            [EnumMember(Value = "D")]
            D = 4,

            /// <summary>
            /// Enum E for value: E
            /// </summary>
            [EnumMember(Value = "E")]
            E = 5,

            /// <summary>
            /// Enum F for value: F
            /// </summary>
            [EnumMember(Value = "F")]
            F = 6,

            /// <summary>
            /// Enum G for value: G
            /// </summary>
            [EnumMember(Value = "G")]
            G = 7,

            /// <summary>
            /// Enum H for value: H
            /// </summary>
            [EnumMember(Value = "H")]
            H = 8,

            /// <summary>
            /// Enum I for value: I
            /// </summary>
            [EnumMember(Value = "I")]
            I = 9

        }


        /// <summary>
        /// The estimated income of the person, where: A &#x3D; &lt; $20, 000; B &#x3D; $20, 000 - $29, 999; C &#x3D; $30, 000 - $39, 999; D &#x3D; $40, 000 - $49, 999; E &#x3D; $50, 000 - $74, 999; F &#x3D; $75, 000 - $99, 999; G &#x3D; $100, 000 - $124, 999; H &#x3D; $125, 000 - $149, 999; I &#x3D; &gt; $149,999.
        /// </summary>
        /// <value>The estimated income of the person, where: A &#x3D; &lt; $20, 000; B &#x3D; $20, 000 - $29, 999; C &#x3D; $30, 000 - $39, 999; D &#x3D; $40, 000 - $49, 999; E &#x3D; $50, 000 - $74, 999; F &#x3D; $75, 000 - $99, 999; G &#x3D; $100, 000 - $124, 999; H &#x3D; $125, 000 - $149, 999; I &#x3D; &gt; $149,999.</value>
        [DataMember(Name = "estimatedIncome", EmitDefaultValue = false)]
        public EstimatedIncomeEnum? EstimatedIncome { get; set; }
        /// <summary>
        /// The estimated net worth of the person, where: A &#x3D; &lt; $5k; B &#x3D; $5k - $19k; C &#x3D; $20k - $49k; D &#x3D; $50k - $80k; E &#x3D; $81k - $99k; F &#x3D; $100k - $249k; G &#x3D; $250k - $499k; H &#x3D; &gt; $500k.
        /// </summary>
        /// <value>The estimated net worth of the person, where: A &#x3D; &lt; $5k; B &#x3D; $5k - $19k; C &#x3D; $20k - $49k; D &#x3D; $50k - $80k; E &#x3D; $81k - $99k; F &#x3D; $100k - $249k; G &#x3D; $250k - $499k; H &#x3D; &gt; $500k.</value>
        [JsonConverter(typeof(StringEnumConverter))]
        public enum EstimatedNetWorthEnum
        {
            /// <summary>
            /// Enum A for value: A
            /// </summary>
            [EnumMember(Value = "A")]
            A = 1,

            /// <summary>
            /// Enum B for value: B
            /// </summary>
            [EnumMember(Value = "B")]
            B = 2,

            /// <summary>
            /// Enum C for value: C
            /// </summary>
            [EnumMember(Value = "C")]
            C = 3,

            /// <summary>
            /// Enum D for value: D
            /// </summary>
            [EnumMember(Value = "D")]
            D = 4,

            /// <summary>
            /// Enum E for value: E
            /// </summary>
            [EnumMember(Value = "E")]
            E = 5,

            /// <summary>
            /// Enum F for value: F
            /// </summary>
            [EnumMember(Value = "F")]
            F = 6,

            /// <summary>
            /// Enum G for value: G
            /// </summary>
            [EnumMember(Value = "G")]
            G = 7,

            /// <summary>
            /// Enum H for value: H
            /// </summary>
            [EnumMember(Value = "H")]
            H = 8

        }


        /// <summary>
        /// The estimated net worth of the person, where: A &#x3D; &lt; $5k; B &#x3D; $5k - $19k; C &#x3D; $20k - $49k; D &#x3D; $50k - $80k; E &#x3D; $81k - $99k; F &#x3D; $100k - $249k; G &#x3D; $250k - $499k; H &#x3D; &gt; $500k.
        /// </summary>
        /// <value>The estimated net worth of the person, where: A &#x3D; &lt; $5k; B &#x3D; $5k - $19k; C &#x3D; $20k - $49k; D &#x3D; $50k - $80k; E &#x3D; $81k - $99k; F &#x3D; $100k - $249k; G &#x3D; $250k - $499k; H &#x3D; &gt; $500k.</value>
        [DataMember(Name = "estimatedNetWorth", EmitDefaultValue = false)]
        public EstimatedNetWorthEnum? EstimatedNetWorth { get; set; }
        /// <summary>
        /// Initializes a new instance of the <see cref="DemographicsAppendResponseDemographicsPersonFinances" /> class.
        /// </summary>
        /// <param name="estimatedIncome">The estimated income of the person, where: A &#x3D; &lt; $20, 000; B &#x3D; $20, 000 - $29, 999; C &#x3D; $30, 000 - $39, 999; D &#x3D; $40, 000 - $49, 999; E &#x3D; $50, 000 - $74, 999; F &#x3D; $75, 000 - $99, 999; G &#x3D; $100, 000 - $124, 999; H &#x3D; $125, 000 - $149, 999; I &#x3D; &gt; $149,999..</param>
        /// <param name="estimatedNetWorth">The estimated net worth of the person, where: A &#x3D; &lt; $5k; B &#x3D; $5k - $19k; C &#x3D; $20k - $49k; D &#x3D; $50k - $80k; E &#x3D; $81k - $99k; F &#x3D; $100k - $249k; G &#x3D; $250k - $499k; H &#x3D; &gt; $500k..</param>
        /// <param name="isCharityDonar">Indicates whether the person is flagged as a charity donar..</param>
        /// <param name="usesCreditCard">Indicates whether the person is flagged as a credit card user..</param>
        public DemographicsAppendResponseDemographicsPersonFinances(EstimatedIncomeEnum? estimatedIncome = default(EstimatedIncomeEnum?), EstimatedNetWorthEnum? estimatedNetWorth = default(EstimatedNetWorthEnum?), bool isCharityDonar = default(bool), bool usesCreditCard = default(bool))
        {
            this.EstimatedIncome = estimatedIncome;
            this.EstimatedNetWorth = estimatedNetWorth;
            this.IsCharityDonar = isCharityDonar;
            this.UsesCreditCard = usesCreditCard;
        }

        /// <summary>
        /// Indicates whether the person is flagged as a charity donar.
        /// </summary>
        /// <value>Indicates whether the person is flagged as a charity donar.</value>
        [DataMember(Name = "isCharityDonar", EmitDefaultValue = true)]
        public bool IsCharityDonar { get; set; }

        /// <summary>
        /// Indicates whether the person is flagged as a credit card user.
        /// </summary>
        /// <value>Indicates whether the person is flagged as a credit card user.</value>
        [DataMember(Name = "usesCreditCard", EmitDefaultValue = true)]
        public bool UsesCreditCard { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append("class DemographicsAppendResponseDemographicsPersonFinances {\n");
            sb.Append("  EstimatedIncome: ").Append(EstimatedIncome).Append("\n");
            sb.Append("  EstimatedNetWorth: ").Append(EstimatedNetWorth).Append("\n");
            sb.Append("  IsCharityDonar: ").Append(IsCharityDonar).Append("\n");
            sb.Append("  UsesCreditCard: ").Append(UsesCreditCard).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return Newtonsoft.Json.JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as DemographicsAppendResponseDemographicsPersonFinances);
        }

        /// <summary>
        /// Returns true if DemographicsAppendResponseDemographicsPersonFinances instances are equal
        /// </summary>
        /// <param name="input">Instance of DemographicsAppendResponseDemographicsPersonFinances to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(DemographicsAppendResponseDemographicsPersonFinances input)
        {
            if (input == null)
            {
                return false;
            }
            return 
                (
                    this.EstimatedIncome == input.EstimatedIncome ||
                    this.EstimatedIncome.Equals(input.EstimatedIncome)
                ) && 
                (
                    this.EstimatedNetWorth == input.EstimatedNetWorth ||
                    this.EstimatedNetWorth.Equals(input.EstimatedNetWorth)
                ) && 
                (
                    this.IsCharityDonar == input.IsCharityDonar ||
                    this.IsCharityDonar.Equals(input.IsCharityDonar)
                ) && 
                (
                    this.UsesCreditCard == input.UsesCreditCard ||
                    this.UsesCreditCard.Equals(input.UsesCreditCard)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                hashCode = (hashCode * 59) + this.EstimatedIncome.GetHashCode();
                hashCode = (hashCode * 59) + this.EstimatedNetWorth.GetHashCode();
                hashCode = (hashCode * 59) + this.IsCharityDonar.GetHashCode();
                hashCode = (hashCode * 59) + this.UsesCreditCard.GetHashCode();
                return hashCode;
            }
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        public IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}